app/Kernel.php

```diff
@@ -225,6 +228,50 @@ public function getCacheDir(): string
         return dirname(__DIR__) . '/var/cache/' . $this->environment . '/' . $this->request->getHost();
     }
 
+    private function setLocaleOnRequest(ContainerBuilder $container): void
+    {
+        $chunks = (array) explode('/', trim($this->request->getRequestUri(), '/'));
+        $defaultLocale = $container->getParameter('site.default_language');
+        $currentLocale = $chunks[0] ?? $defaultLocale;
+
+        $this->request->setLocale(strlen($defaultLocale) === 2 ? $currentLocale : $defaultLocale);
+        $this->request->setDefaultLocale($defaultLocale);
+    }
+
+    private function getSiteDomainForCurrentLocale(ContainerBuilder $container): string
+    {
+        if ($container->hasParameter('site.domain')) {
+            return $container->getParameter('site.domain');
+        }
+
+        $this->setLocaleOnRequest($container);
+
+        if ($container->hasParameter('site.domain.' . $this->request->getLocale())) {
+            return $this->verifyDomainMatchesLocale(
+                $container->getParameter('site.domain.' . $this->request->getLocale())
+            );
+        }
+
+        return $this->verifyDomainMatchesLocale(
+            $container->getParameter('site.domain.' . $this->request->getDefaultLocale())
```

Comment:
What if there is no default locale defined? Or will this always be the case?

Reply:
that will always be the case, site.domain is there by default